torch/export/_trace.py

```diff
@@ -1423,12 +1426,57 @@ def _make_fx_helper(mod, args, kwargs, **flags):
         def wrapped_fn(*args):
             return tuple(flat_fn(*args))
 
+        buffers_to_mutate = {}
         with enable_python_dispatcher():
-            gm = make_fx(
-                wrapped_fn,
-                record_module_stack=True,
-                pre_dispatch=True,
-            )(*flat_args)
+            detect_attribute_assignment_ctx = nullcontext()
+            if hasattr(mod, "_export_root"):
+                detect_attribute_assignment_ctx = _detect_attribute_assignment(  # type: ignore[assignment]
+                    mod._export_root
+                )
+
+                assigned_buffers = {}
+
+                def _map_assigned_buffer_to_proxy(_mod, name, buffer):
```

Comment:
Could we make this a util function in aot_autograd.py and use that here? I think this logic also exist in aot_autograd.py?

Reply:
Mostly true, although there we are seeing buffers as Functional(Fake(...)) and here just Fake(...).

Reply:
Could we filter out the other parts? What i am worried is this part is bit complicated to understand so it would be better if we have only place to change stuff. Maybe the util function can take what instance we want to assert on lol.